examples/plot_gp_function.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[ ]:
+
+import numpy as np
+
+import matplotlib.pyplot as plt
+from matplotlib import gridspec
+import sys
+import os 
+# current working directory
+path = os.getcwd()
+# parent directory
+parent = os.path.join(path, os.pardir)
+sys.path.append(os.path.abspath(parent))
+from bayes_opt1 import BayesianOptimization
+from bayes_opt1 import UtilityFunction
+
+
+def normalization(utility):
+    return (utility- min(utility)) / (max(utility) - min(utility))
+
+
+def plot_gp(optimizer1, optimizer2, optimizer3, x, target, params):
```

Comment:
Vi suggerisco un cambiamento in vista dell'integrazione con KG. I primi tre argomenti potrebbero diventare un'unica lista/dizionario di oggetti chiamata `optimizers`, su cui poi potreste operare dei loop. Parlo di dizionario perché potreste mappare le stringhe dei nomi delle acquisizioni agli oggetti optimizer. Per esempio alcuni comandi diventerebbero:
```python
x_obs = {}  # dizionario vuoto
for acq, optimizer in optimizers.items():  # acq è la stringa-chiave, optimizer è l'oggetto
  x_obs[acq] = np.array([[res["params"]["x"]] for res in optimizer.res])
  ...
```
Insomma, fareste le stesse cose di prima ma usando liste/dizionari invece di scrivere manualmente le diverse variabili per ogni acquisizione. Questo permetterebbe di plottare un arbitrario numero di diverse acquisizioni

Reply:
Per quanto riguarda i colori, mi sembra che matplotlib ne assegni automaticamente uno diverso ad ogni cosa nuova che si plotta, quindi non dovreste avere problemi. Se proprio vi serve, potete definire una colormap che contiene `n` colori diversi con:
```python
n = len(optimizers)
colmap = plt.cm.get_cmap(name='rainbow', lut=n)
```
e scrivere `colmap(i)` quando vi serve l'`i`-esimo colore